core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletFile.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.metadata.schema;
+
+import org.apache.accumulo.core.data.TableId;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Text;
+
+/**
+ * Object representing a tablet file entry in the metadata table. Keeps a string of the exact entry
+ * of what is in the metadata table for the column qualifier of the
+ * {@link org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.DataFileColumnFamily}
+ * Validates the full URI form: "hdfs://1.2.3.4/accumulo/tables/2a/t-0003/C0004.rf"
+ */
+public class TabletFile implements Comparable<TabletFile> {
+  private String volume; // hdfs://1.2.3.4/accumulo
+  // /tables/
+  private TableId tableId; // 2a
+  private String tabletDir; // t-0003
+  private String fileName; // C0004.rf
+
+  // exact string that is stored the metadata table
+  private String metadataEntry;
+  private Path metadataPath;
+
+  public TabletFile(String metadataEntry) {
```

Comment:
Older relative paths stored in the metadata table do not have a table ID in the value and this may come from the row.  I was thinking this class could be a pure POJO w/o any code for serialization and deserialization, with the goal being to consolidate this type of code in one place inside Ample.  Currently the serialization and deserialization code is spread far and wide.  As intermediate step could have a static method for now of the form `public static TabletFile newTableFile(Key k, Value v)` and maybe this could be placed in TabletFileUtil for now.  This static method could support the older relative paths.

Reply:
I think I saw a test hanging due to the relative path in VolumeIT.  Thankfully, it looks like we do have a test for the older paths with VolumeIT.testRelativePaths()